packages/data-access/specs/request-block-specification-v0.1.0.md

```diff
@@ -0,0 +1,80 @@
+# Request Block Specification v0.1.0
+
+You can be interested in this document if:
+
+- You want to create your own implementation of the request protocol
+- You are curious enough to dive and see what is under the hood of the request protocol
+
+You don't need to read this document if:
+
+- You want to develop an app using the request protocol (see the API library instead [here](/packages/request-client.js))
+
+The Data Access layer is responsible for:
+
+- Indexing transactions to allow retrieval
+- Batching transactions
+
+## Transactions indexing (Channel & Topics)
+
+The indexing of the transactions is made by two mechanisms:
+
+- Channel
+- Topics
+
+The channel in data-access allows to group several transactions under one string: the channel id. This channel is indexed by several strings called topics.
+
+There are two ways of getting transactions:
+
+- by channel id: this gets all the transactions of the channel
+- by topic: this gets all the transactions of the channels indexed by this topic
+
+## Transaction batching
+
+The transaction are stored and indexed in blocks.
+A block is JSON object containing two main parts:
+
+- header: containing the information to index the transactions (channels and topics)
+- transactions: the transactions themselves
+
+## Block Properties
+
+| Property              | Type   | Description                                                     |
+| --------------------- | ------ | --------------------------------------------------------------- |
+| **header.channelIds** | Object | Array of positions in transactions Array indexed by channel ids |
+| **header.topics**     | Object | Array of topics indexed by channel ids                          |
```

Comment:
Mhh, it's rather the channel id that is indexed by a topic not the opposite, isn't it ?

Reply:
nope it is not :)